src/sparc/client/client.py

```diff
@@ -69,39 +67,52 @@ def __init__(self, config_file="config/config.ini", connect=True) -> None:
                 f"{__package__}.services.{module_name}", config[current_config], connect
             )
 
-    def add_module(self, path, config, connect=True):
+    #        else:
+    #            self.add_module(package_dir, config[current_config], connect)
+
+    def add_module(self, paths, config=None, connect=True):
```

Comment:
IMO annotating the function signatures, specially in base classes, is really important. 
```python
def add_module(self, 
   paths: Union[str, list[str]], 
   config: Optional[Union[dict, configparser.SectionProxy]]=None, 
   connect: bool=True,
  ) -> None:
     ...
```
It is not only self documenting but we can also use type checkers like [mypy](https://www.mypy-lang.org/) to validate our code. Moreover, these annotations will never be outdated since they are part of the code.

Reply:
Interesting idea @pcrespov . Should we use static functon signatures in this project?
Never did it in my code, but happy to use them. What do you think @nih-sparc/sparc-tech-leads ?

Reply:
I have added signatures in one of the commits.